src/ucp/api/ucp.h

```diff
@@ -1416,6 +1417,15 @@ typedef struct {
      * which means the memory type will be detected internally.
      */
     ucs_memory_type_t memory_type;
+
+    /**
+     * Pointer to the information where received data details are stored
+     * in case of immediate completion of receive operation.
+     */
+    union {
+        size_t              *length;
```

Comment:
Can you please add description for each one of the fields. I guess user has to allocate the memory and then it is updated in function return ?

Reply:
ok

Reply:
added